utils/streams.js

```diff
@@ -0,0 +1,205 @@
+  cliRun();
+}
+
+function cliRun() {
+  const argv = Argv
+    .version('1.0.0')
+    .usage('Usage: $0 --action actionType [--file filePath]')
+    .example('$0 --action=io --file=./src/example.js')
+    .example('$0 --action transform --file=./data/test.txt')
+    .example('$0 -a csvToJsonStream -f ./data/test.csv')
+    .example('$0 -a csvToJsonFile -f ./data/test.csv')
+    .example('$0 --action cssBundler --path ./data/css')
+    .demandOption(['action'])
+    .option('action', {
+      alias: 'a',
+      describe: 'choose an action',
+      choices: ['io', 'transform', 'csvToJsonStream', 'csvToJsonFile', 'cssBundler', 'generateTestCSS']
+    })
+    .option('file', {
+      alias: 'f',
+      describe: 'provide a path to file (mandatory for -a=csvToJsonFile or -a=io)'
+    })
+    .option('path', {
+      alias: 'p',
+      describe: 'provide a path to folder to bundle CSS files (mandatory for --action=cssBundler)'
+    })
+    .help('h')
+    .argv;
+  
+  if (argv.action) {
+    if(argv.action === 'cssBundler') actions[argv.action](argv.path);
+    else actions[argv.action](argv.file);
+  }
+}
+
+/* stream helpers*/
+function inputOutput(filePath) { /* 4, babel-node ./utils/streams.js -a io -f ./data/test.txt */
+  if (fs.existsSync(filePath)) fs.createReadStream(filePath).pipe(process.stdout);
+  else console.log('Please specify correct file path');
+}
+
+function stdInInformUser() {
+  process.stdout.write(`Please start typing to transform ...
+  to end input use CTRL + D for macOS and Alt-Space. Then, C for Windows\n`);
+  process.stdin.setEncoding('utf8');
+}
+
+function end() {
+  process.stdout.write('Transform was completed\n');
+}
+
+function write(buffer, encoding, next) {
+  this.push(`Transformed to: ${buffer.toString().toUpperCase()}`);
+  next();
+}
+
+function transform() { /* 5, node ./utils/streams.js -a transform */
+  stdInInformUser();
+  process.stdin
+    .pipe(through(write, end))
+    .pipe(process.stdout);
+}
+
+function csvToJsonStreamTransform(buffer, encoding, next) {
+  let options = {
+    delimiter : ',',
+    quote     : '"'
+  };
+  this.push(JSON.stringify(csvjson.toObject(buffer.toString(), options)));
+  next();
+}
+
+function csvToJsonStream(filePath) { /* 6, babel-node ./utils/streams.js -a csvToJsonStream -f ./data/test.csv */
+  if (fs.existsSync(filePath) && path.extname(filePath) === '.csv') {
+    const readableStream = fs.createReadStream(filePath, 'utf8');
+    readableStream
+      .pipe(through(csvToJsonStreamTransform))
+      .pipe(process.stdout);
+  }
+  else console.log('Please specify correct file path');
+}
+
+function createFilePath(filePath, extension = '.json', name) {
+  const pathJsonFile = path.parse(filePath);
+  
+  pathJsonFile.ext = extension;
+  pathJsonFile.base = `${pathJsonFile.name || name}${pathJsonFile.ext}`;
+  return path.format(pathJsonFile);
+}
+
+function csvToJsonFile(filePath) { /* 7, babel-node ./utils/streams.js -a csvToJsonFile -f ./data/test.csv */
```

Comment:
`cssToJsonFile` and `csvToJsonStream` are almost the same. I guess you can create one function with optional argument `outputFile`.

And at the moment you still don't transform csv to json correctly.

Reply:
thx, fixed